src/plugins/intel_cpu/src/emitters/x64/jit_eltwise_emitters.cpp

```diff
@@ -107,76 +99,129 @@ void jit_mul_add_emitter::emit_isa(const std::vector<size_t> &in_vec_idxs, const
+                } else if (vmm_dst.getIdx() == vmm_src2.getIdx()) {
+                    Vmm vmm_aux0 = Vmm(aux_vec_idxs[0]);
+                    h->uni_vpmulld(vmm_aux0, vmm_src0, vmm_src1);
+                    h->uni_vpaddd(vmm_dst, vmm_dst, vmm_aux0);
+                } else {
+                    h->uni_vpmulld(vmm_dst, vmm_src0, vmm_src1);
+                    h->uni_vpaddd(vmm_dst, vmm_dst, vmm_src2);
+                }
+            } break;
+            case Precision::I64: {
+                Vmm vmm_aux0 = Vmm(aux_vec_idxs[0]);
+                Vmm vmm_aux1 = Vmm(aux_vec_idxs[1]);
+                // There is no multiply int64 instruction on AVX2 and SSE41, thus the WA is used.
+                // vmm_src0 = ab; vmm_src1 = cd;
+                h->uni_vpsrlq(vmm_aux0, vmm_src0, 32);
+                h->uni_vpmuludq(vmm_aux0, vmm_aux0, vmm_src1); // a * d
+                h->uni_vpsrlq(vmm_aux1, vmm_src1, 32);
+                h->uni_vpmuludq(vmm_aux1, vmm_aux1, vmm_src0); // b * c
+                h->uni_vpaddq(vmm_aux1, vmm_aux1, vmm_aux0);   // a * d + b * c
+                h->uni_vpsllq(vmm_aux1, vmm_aux1, 32);
+                h->uni_vpmuludq(vmm_aux0, vmm_src0, vmm_src1); // b * d
+                h->uni_vpaddq(vmm_aux0, vmm_aux0, vmm_aux1);   // (a * d + b * c) << 32 + b * d
+
+                h->uni_vpaddq(vmm_dst, vmm_src2, vmm_aux0);
             } break;
-            default: assert(!"unsupported precision");
+            default: IE_THROW() << "jit_mul_add_emitter doesn't support precision '" << exec_prc_ << "'";
         }
     };
 
-    auto uni_vfmadd231_vmm = [this, vmm_aux0](Vmm vmm_dst, Vmm vmm_src0, Vmm vmm_src1, Vmm vmm_src2) {
+    auto uni_madd_vmm = [this](const Vmm &vmm_dst, const Vmm &vmm_src0, const Vmm &vmm_src1, const Vmm &vmm_src2) {
         switch (exec_prc_) {
             case Precision::FP32: {
-                Vmm vmm_mul0;
                 if (vmm_dst.getIdx() == vmm_src0.getIdx()) {
-                    h->uni_vmovups(vmm_aux0, vmm_src0);
-                    vmm_mul0 = vmm_aux0;
+                    h->uni_vfmadd132ps(vmm_src0, vmm_src2, vmm_src1);
+                } else if (vmm_dst.getIdx() == vmm_src1.getIdx()) {
+                    h->uni_vfmadd132ps(vmm_src1, vmm_src2, vmm_src0);
+                } else if (vmm_dst.getIdx() == vmm_src2.getIdx()) {
+                    h->uni_vfmadd231ps(vmm_src2, vmm_src0, vmm_src1);
                 } else {
-                    vmm_mul0 = vmm_src0;
+                    h->uni_vmovups(vmm_dst, vmm_src2);
+                    h->uni_vfmadd231ps(vmm_dst, vmm_src0, vmm_src1);
                 }
-
-                Vmm vmm_mul1;
-                if (vmm_dst.getIdx() == vmm_src1.getIdx()) {
-                    h->uni_vmovups(vmm_aux0, vmm_src1);
-                    vmm_mul1 = vmm_aux0;
+            } break;
+            case Precision::I32: {
+                if (vmm_dst.getIdx() == vmm_src2.getIdx()) {
+                    Vmm vmm_aux0 = Vmm(aux_vec_idxs[0]);
```

Comment:
We use aux register here. Does it mean that we need to return `1` from `aux_vecs_count()` if `exec_prc_ == I32`?